src/client.rs

```diff
@@ -94,9 +94,21 @@ where
      */
     pub async fn with_data_package(
         url: &str,
-        games: Option<Vec<String>>,
+       mut games: Option<Vec<String>>,
     ) -> Result<ArchipelagoClient<S>, ArchipelagoError> {
         let mut client = Self::new(url).await?;
+        if games.is_none() {
+            // If None, request the games that are part of the connected room.
```

Comment:
[The procotocol docs](https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#getdatapackage) imply that if games isn't specified the server will send the data package for every game. Either way, given that the server already declares support for a data package with no games listed, I don't think we should change the behavior in the client.